DylibBinder/XmlGenerator.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Linq;
+using System.Xml;
+
+namespace DylibBinder {
+	public class XmlGenerator {
```

Comment:
Better class. But you have a XmlWriter, that is 'IAsyncDisposable, IDisposable' so you need to Dispose it. Since you need to dispose it, your XmlGenerator is probably going to need to be IDisposable